core/sdk-core/src/test/java/software/amazon/awssdk/core/internal/progress/listener/ProgressUpdaterTest.java

```diff
@@ -0,0 +1,341 @@
+
+        SdkRequest sdkRequest = NoopTestRequest.builder()
+                                               .overrideConfiguration(overrideConfig)
+                                               .build();
+
+        ProgressUpdater progressUpdater = new ProgressUpdater(sdkRequest, null);
+        progressUpdater.responseHeaderReceived();
+
+        assertEquals(0.0, progressUpdater.downloadProgress().progressSnapshot().transferredBytes(), 0.0);
+        Mockito.verify(mockListener, never()).executionFailure(ArgumentMatchers.any(ProgressListener.Context.ExecutionFailure.class));
+        Mockito.verify(mockListener, never()).attemptFailure(ArgumentMatchers.any(ProgressListener.Context.AttemptFailure.class));
+        Mockito.verify(mockListener, never()).attemptFailureResponseBytesReceived(ArgumentMatchers.any(ProgressListener.Context.AttemptFailureResponseBytesReceived.class));
+        Mockito.verify(mockListener, times(1)).responseHeaderReceived(ArgumentMatchers.any(ProgressListener.Context.ResponseHeaderReceived.class));
+
+    }
+
+    @Test
+    public void test_responseBytesReceived_transferredBytes() {
+
+        CaptureProgressListener mockListener = Mockito.mock(CaptureProgressListener.class);
+
+        SdkRequestOverrideConfiguration.Builder builder = SdkRequestOverrideConfiguration.builder();
+        builder.progressListeners(Arrays.asList(mockListener, captureProgressListener));
+
+        SdkRequestOverrideConfiguration overrideConfig = builder.build();
+
+        SdkRequest sdkRequest = NoopTestRequest.builder()
+                                               .overrideConfiguration(overrideConfig)
+                                               .build();
+
+        ProgressUpdater progressUpdater = new ProgressUpdater(sdkRequest, null);
+        progressUpdater.incrementBytesReceived(BYTES_TRANSFERRED);
+        assertEquals(BYTES_TRANSFERRED, progressUpdater.downloadProgress().progressSnapshot().transferredBytes(), 0.0);
+
+        progressUpdater.incrementBytesReceived(BYTES_TRANSFERRED);
+        assertEquals(BYTES_TRANSFERRED + BYTES_TRANSFERRED, progressUpdater.downloadProgress().progressSnapshot().transferredBytes(), 0.0);
+
+        Mockito.verify(mockListener, never()).executionFailure(ArgumentMatchers.any(ProgressListener.Context.ExecutionFailure.class));
+        Mockito.verify(mockListener, never()).attemptFailure(ArgumentMatchers.any(ProgressListener.Context.AttemptFailure.class));
+        Mockito.verify(mockListener, never()).attemptFailureResponseBytesReceived(ArgumentMatchers.any(ProgressListener.Context.AttemptFailureResponseBytesReceived.class));
+        Mockito.verify(mockListener, times(2)).responseBytesReceived(ArgumentMatchers.any(ProgressListener.Context.ResponseBytesReceived.class));
+
+    }
+
+    @ParameterizedTest(name = "{index} - {1}, total bytes = {0}")
+    @MethodSource("contentLength")
+    void registerCompletion_differentTransferredByteRatios_alwaysCompletesOnce(Long givenContentLength)
```

Comment:
Question: what is this behavior this test trying to verify?

Reply:
That the asynchronous request works as intended for multiple content lengths. By intended, it is meant that "executionSuccess" is called once.